File: deliveryRoute/src/main/java/com/sparta/deliveryroute/application/client/HubService.java
package com.sparta.deliveryroute.application.client;

import com.sparta.deliveryroute.application.dto.RouteResult;
import com.sparta.deliveryroute.application.dto.response.HubRouteResponseDto;

import java.util.List;
import java.util.UUID;

public interface HubService {

	RouteResult findHubRoutesById(UUID startHubId, UUID endHubId);

	List<HubRouteResponseDto> findHubRoutesByHubRouteId(List<UUID> hubRouteIdList);
}

File: hub/src/main/kotlin/com/sparta/hub/domain/service/HubService.kt
package com.sparta.hub.domain.service

import com.sparta.hub.application.dto.RouteResult
import com.sparta.hub.application.dto.request.HubRequestDto
import com.sparta.hub.application.dto.request.HubSearchRequestDto
import com.sparta.hub.application.dto.response.HubDetailResponseDto
import com.sparta.hub.application.dto.response.HubSummaryResponseDto
import org.springframework.data.domain.Page
import org.springframework.data.domain.Pageable
import java.util.*

interface HubService {

    fun registerHub(hubAddress: String, hubName: String): UUID

    fun navigateHubRoutes()

    fun getOptimalHubRoutes(startHubName: String, endHubName: String): RouteResult

    fun getHubs(pageable: Pageable, requestDto: HubSearchRequestDto): Page<HubSummaryResponseDto>

    fun getHubDetail(hubId: UUID): HubDetailResponseDto

    fun modifyHub(hubId: UUID, hubRequestDto: HubRequestDto): UUID

    fun existHub(hubId: UUID): Boolean
}
